staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go

```diff
@@ -272,29 +271,41 @@ func (s *singleBenchmark) run(b *testing.B) {
 }
 
 func (s *singleBenchmark) bench(b *testing.B) {
-	s.b = b
+	// Simulate slowness, qps limit, external service limitation, etc
+	const maxInFlight = 40
+	chokepoint := make(chan struct{}, maxInFlight)
+	// lookup count
+	var lookups uint64
+
 	a := newWithClock(
-		authenticator.TokenFunc(s.lookup),
+		authenticator.TokenFunc(func(ctx context.Context, token string) (*authenticator.Response, bool, error) {
+			atomic.AddUint64(&lookups, 1)
+
+			chokepoint <- struct{}{}
+			defer func() { <-chokepoint }()
+
+			time.Sleep(1 * time.Millisecond)
+
+			return s.lookup(ctx, token)
+		}),
 		true,
 		4*time.Second,
 		500*time.Millisecond,
 		utilclock.RealClock{},
 	)
-	const maxInFlight = 40
-	s.chokepoint = make(chan struct{}, maxInFlight)
-
-	s.b.ResetTimer()
 
+	b.ResetTimer()
 	b.SetParallelism(s.threadCount)
 	b.RunParallel(func(pb *testing.PB) {
 		r := mathrand.New(mathrand.NewSource(mathrand.Int63()))
 		for pb.Next() {
-			// some problems appear with random
-			// access, some appear with many
-			// requests for a single entry, so we
-			// do both.
-			s.doAuthForTokenN(r.Intn(len(s.tokens)), a)
+			s.doAuthForTokenN(r.Intn(s.tokenCount), a)
+			// some problems appear with random access, some appear with many
+			// requests for a single entry, so we do both.
```

Comment:
Uh, did you mean to drop the `s.doAuthForTokenN(r.Intn(len(s.tokens)), a)`?

Reply:
I reordered it to before the comment on line 302, because it seemed like it was explaining the second doAuthForTokenN but reading it again, it looks like it was in the right spot.

Reply:
Fixed.